pymemcache/client/base.py

```diff
@@ -817,6 +817,37 @@ def quit(self):
         self._misc_cmd([cmd], b'quit', True)
         self.close()
 
+    def shutdown(self, graceful=False):
+        """
+        The memcached "shutdown" command.
+
+        This will request shutdown and eventual termination of the server,
+        optionally preceded by a graceful stop of memcached's internal state
+        machine. Note that the server needs to have been started with the
+        shutdown protocol command enabled with the --enable-shutdown flag.
+
+        Args:
+          graceful: optional bool, True to request a graceful shutdown with
+                    SIGUSR1 (defaults to False, i.e. SIGINT shutdown).
+
+        Returns:
+          True.
+        """
+        cmd = b'shutdown'
+        if graceful:
+            cmd += b' graceful'
+        cmd += b'\r\n'
+
+        # The shutdown command raises a server-side error if the shutdown
+        # protocol command is not enabled. Otherwise, a successful shutdown
+        # is expected to close the remote end of the transport.
+        try:
+            self._misc_cmd([cmd], b'shutdown', False)
+        except MemcacheUnexpectedCloseError:
+            pass
+
+        return True
```

Comment:
Perhaps this function should not return anything (`None` by default), which would match `quit`'s behavior.

Reply:
I suppose it could also return `False` if `MemcacheUnknownCommandError` is raised, but I think it's better to surface that error directly to the caller.